src/java_bytecode/java_object_factory.cpp

```diff
@@ -389,6 +389,54 @@ void java_object_factoryt::gen_pointer_target_init(
   }
 }
 
+/// Recursion-set entry owner class. If a recursion-set entry is added
+/// in a particular scope, ensures that it is erased on leaving
+/// that scope.
+class recursion_set_entryt
```

Comment:
Any reason not to just stick in its own file, given how huge this one already is?